docs/design/coreclr/botr/logging.md

```diff
@@ -0,0 +1,70 @@
+Runtime logging for developers
+==============================
+
+Date: Feb. 2024
+
+The .NET runtime codebase is massive, spread across thousands of files in multiple languages. There are thousands of log messages spread throughout that codebase, and most of them are disabled by default. So how do you, a runtime developer beset by all manner of mysterious test failures, get those messages out of the runtime and into your console or log files? And how should you go about adding log messages appropriately to new code you're writing?
+
+*NOTE: This document will repeatedly reference many environment variables. For a detailed list of environment variables supported by the runtime and explanations of what they do, consult [`clrconfigvalues.h`](https://github.com/dotnet/runtime/blob/main/src/coreclr/inc/clrconfigvalues.h), [`gcconfig.h`](https://github.com/dotnet/runtime/blob/main/src/coreclr/gc/gcconfig.h), and [`jitconfigvalues.h`](https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/jitconfigvalues.h).*
+
+Types of logging
+================
+
+ETW / EventPipe logging
+-----------------------
+"Event Tracing for Windows", or ETW, is an extensible tracing system built into Windows that various applications and services can use to report events in a way that allows the end user to filter and record them dynamically. ETW is consumed by tools like perfview, xperf, UIforETW, or WPR/WPA. This system is frequently used by customers to understand what is happening in development or production environments.
+
+EventPipe is a cross-platform tracing system similar to ETW that is used widely in the .NET runtime.
+For a detailed overview of EventPipe, see [the EventPipe documentation pages on .NET Learn](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/eventpipe). It offers excellent performance along with lots of flexibility, so it is one of the main ways to get information from an active runtime that we advertise. Most events that we support in debug builds also are exposed in release builds.
+
+For basic scenarios, you can use the [`dotnet-counters`](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-counters) tool to monitor events and performance counters the runtime reports through EventPipe. For example, to collect the default counters from a project while running it, you could use `dotnet-counters collect -- dotnet run`.
+
+You can also use the [`dotnet-trace`](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/dotnet-trace) tool to capture EventPipe events from the runtime on the fly, either saving them to a file or sending them to your console. It provides command line options you can use to filter events by severity level or based on keywords. For example, to capture high-severity GC events only from a project while running it, you could use `dotnet-trace collect --clreventlevel error --clrevents gc -- dotnet run`.
```

Comment:
there's no error level events for GC - if you look at clretwall.man, there's only one error level event for exception. 

also just checking if this is intentional - with this command line you'd be profiling the dotnet process itself instead of the process it runs. since this is done in the botr context, the former could be intentional but since you mentioned "from a project while running it", it sounds like you might mean the latter.